### PR TITLE
Feature promiscuous

### PR DESCRIPTION
**Pull request by rakshasa**
Adds promiscuous interface mode, although it relies on a hack so it only supports packets from remote nodes.

### CODE REVIEW

File: vnet/lib/vnet/openflow/ovs_ofctl.rb [hunk 1]
```diff
@@ -37,20 +37,36 @@ def add_flow(flow)
       debug log_format("'#{command}' => #{result}") if verbose
     end
 
+    # TODO: Deprecate.
```
Comment (Metallion) [inline vnet/lib/vnet/openflow/ovs_ofctl.rb]: Why not deprecate right away by putting a big ugly message in the log? Something like this:

```ruby
logger.warn "######################################################\n"\
            "add_ovs_flow is deprecated! Use add_ovs_flow2 instead!\n"\
            "######################################################\n\n"\
            "add_ovs_flow was used here: #{caller.first}"                   
```

File: vnet/lib/vnet/openflow/ovs_ofctl.rb [hunk 2]
```diff
@@ -37,20 +37,36 @@ def add_flow(flow)
...
     def add_ovs_10_flow(flow_str)
       command = "#{@ovs_ofctl_10} add-flow #{switch_name} #{flow_str}"
       result = system(command)
 
       debug log_format("'#{command}' => #{result}") if verbose
     end
 
+    def add_ovs_flow2(flow_str)
```
Comment (Metallion) [inline vnet/lib/vnet/openflow/ovs_ofctl.rb]: I assume a code-wide search and replace will be done to rename this to `add_ovs_flow` once all the usage of the former method has been eliminated?

File: integration_test/lib/vnspec/spec/promiscuous_seg_spec.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  #     expect(vm7).to be_able_to_ping(to_gw, 10)
+  #   end
+
+  #   it 'reaches the internet' do
+  #     to_gw = double()
+  #     allow(to_gw).to receive(:ipv4_address).and_return(config[:pseudo_global_ip])
+
+  #     expect(vm7).to be_able_to_ping(to_gw, 10)
+  #   end
+  # end
```
Comment (rakshasa) [inline integration_test/lib/vnspec/spec/promiscuous_seg_spec.rb]: Fixed.

File: vnet/lib/vnet/openflow/ovs_ofctl.rb (hunk 1 above)
Comment (rakshasa) [inline vnet/lib/vnet/openflow/ovs_ofctl.rb]: Fixed.

File: vnet/lib/vnet/openflow/ovs_ofctl.rb (hunk 2 above)
Comment (rakshasa) [inline vnet/lib/vnet/openflow/ovs_ofctl.rb]: Replaced add_ovs_flow.

**Comment by rakshasa** [comment]
Fixed review issues.

**Pull request merged**